### PR TITLE
Add support for PostgreSQL 16

### PR DESCRIPTION
**Pull request by komainu8**
PostgreSQL 16 released at 14th September 2023.

### CODE REVIEW

**Comment by komainu8** [comment]
I will added PostgreSQL 17 snapshot separately.

File: .github/workflows/package.yml [hunk 1]
```diff
@@ -77,6 +77,17 @@ jobs:
       fail-fast: false
       matrix:
         package-id:
+          - postgresql-16-pgdg-pgroonga-debian-bookworm
+          - postgresql-16-pgdg-pgroonga-debian-bookworm-arm64
+          - postgresql-16-pgdg-pgroonga-debian-bullseye
+          - postgresql-16-pgdg-pgroonga-debian-bullseye-arm64
```
Comment (kou) [inline .github/workflows/package.yml:83]: I don't want to add new packages for oldstable...
How about removing them?

File: .github/workflows/package.yml [hunk 2]
```diff
@@ -77,6 +77,17 @@ jobs:
...
+          - postgresql-16-pgdg-pgroonga-ubuntu-focal
+          - postgresql-16-pgdg-pgroonga-ubuntu-focal-arm64
+          - postgresql-16-pgdg-pgroonga-ubuntu-jammy
+          - postgresql-16-pgdg-pgroonga-ubuntu-jammy-arm64
+          - postgresql-16-pgdg-pgroonga-almalinux-9
+          - postgresql-16-pgdg-pgroonga-almalinux-8
+          - postgresql-16-pgdg-pgroonga-centos-7
```
Comment (kou) [inline .github/workflows/package.yml:90]: I don't want to add new packages for CentOS 7...
How about removing this?

Comment (komainu8) [reply .github/workflows/package.yml]: I see. I've removed this.

File: .github/workflows/package.yml (hunk 1 above)
Comment (komainu8) [reply .github/workflows/package.yml]: I see. I've removed them.

**Comment by komainu8** [comment]
@kou 
Thank you for your comments.
I've fixed them.

File: packages/postgresql-16-pgdg-pgroonga/debian/compat [hunk 3]
```diff
@@ -0,0 +1 @@
+13
```
Comment (kou) [inline packages/postgresql-16-pgdg-pgroonga/debian/compat:1]: 12?

File: packages/postgresql-16-pgdg-pgroonga/debian/rules [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+# This has to be exported to make some magic below work.
+export DH_OPTIONS
+
+%:
+	dh $@
+
+override_dh_auto_build:
+	dh_auto_build -- HAVE_MSGPACK=1
```
Comment (kou) [inline packages/postgresql-16-pgdg-pgroonga/debian/rules:13]: Can we add `HAVE_XXHASH=1` too to use system xxHash?

```suggestion
	dh_auto_build -- HAVE_MSGPACK=1 HAVE_XXHASH=1
```

**Comment by kou** [comment]
It seems that `libxxhash-dev` isn't installed.

File: packages/postgresql-16-pgdg-pgroonga/debian/rules [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+export DH_OPTIONS
+
+export XXHASH_MAJOR_VERSION := $(apt-cache show libxxhash-dev | grep Version | awk '{print $$2}' | awk -F '-' '{print $$1}' | awk -F '.' '{print $$1}')
+export XXHASH_MINOR_VERSION := $(apt-cache show libxxhash-dev | grep Version | awk '{print $$2}' | awk -F '-' '{print $$1}' | awk -F '.' '{print $$2}')
+
+%:
+	dh $@
+
+override_dh_auto_build:
+	if [ $(XXHASH_MAJOR_VERSION) >= 0 ] && [ $(XXHASH_MINOR_VERSION) > 7 ]; then
```
Comment (kou) [inline packages/postgresql-16-pgdg-pgroonga/debian/rules:16]: Can we use `if pkg-config --exists "libxxhash >= 0.8.0"; then`?

File: packages/postgresql-16-pgdg-pgroonga/debian/rules [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+%:
+	dh $@
+
+override_dh_auto_build:
+	if pkg-config --exists "libxxhash >= 0.8.0"; then
+		dh_auto_build -- HAVE_MSGPACK=1 HAVE_XXHASH=1
+	else
+		dh_auto_build -- HAVE_MSGPACK=1
+	fi
```
Comment (kou) [inline packages/postgresql-16-pgdg-pgroonga/debian/rules:17]: We need continuous lines that can be added by `C-c C-\` with this region.

**Pull request merged**